### PR TITLE
JAMES-3495 MessageId field in messageIdTable can be null

### PR DESCRIPTION
**Pull request by chibenwa**
Most productive stretching session EVER ;-)

### CODE REVIEW

**Review by rouazana — APPROVED** [review]
nice catch!

**Review by remk — APPROVED** [review]

**Comment by chibenwa** [comment]
test this please

File: mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java [hunk 1]
```diff
@@ -334,7 +335,11 @@ private PreparedStatement prepareSelectUidRangeLimited(Session session) {
     private Optional<ComposedMessageIdWithMetaData> fromRowToComposedMessageIdWithFlags(Row row) {
         if (row.getUUID(MESSAGE_ID) == null) {
             // Out of order updates with concurrent deletes can result in the row being partially deleted
-            // We filter out such records
+            // We filter out such records, and cleanup them.
+            delete(CassandraId.of(row.getUUID(MAILBOX_ID)),
+                MessageUid.of(row.getLong(IMAP_UID)))
+                .subscribeOn(Schedulers.elastic())
+                .subscribe();
```
Comment (rouazana) [inline mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java:342]: thanks! but is it frequent to subscribe to a scheduler inside of a Flux?

Comment (chibenwa) [reply mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAO.java:342]: Yes, we do it eg on the MailQueue to perform some async operations. (We want to trigger it, we do not care about its completion.)

**Review by Arsnael — APPROVED** [review]

**Review by LanKhuat — APPROVED** [review]

**Comment by Arsnael** [comment]
Merged

**Pull request closed without merging**